lib/serviceBusMessage.ts

```diff
@@ -936,7 +936,9 @@ export class ServiceBusMessage implements ReceivedMessage {
    * rejecting the message.
    * @returns Promise<void>
    */
-  async deadLetter(options?: DeadLetterOptions): Promise<void> {
+  async deadLetter(
+    options: DeadLetterOptions = { deadletterReason: "", deadLetterErrorDescription: "" }
+  ): Promise<void> {
     let error: AmqpError = {};
     if (options) {
```

Comment:
With the new method signature I think we can further simplify this code. Options will always be non-empty now, so I think we should always create the error object from the options argument. Depending on what is passed in, we are going to either put an empty string into the error object or actual non-empty strings that the user has specified.